File: 3248. Snake in Matrix.java
class Solution {
    public int finalPositionOfSnake(int n, List<String> commands) {
        int x = 0, y = 0;
        for(int i = 0; i < commands.size(); i++){
            if(commands.get(i).charAt(0) == 'U') x--;
            else if(commands.get(i).charAt(0) == 'D') x++;
            else if(commands.get(i).charAt(0) == 'L') y--;
            else if(commands.get(i).charAt(0) == 'R') y++;
        }
        
        return (x * n) + y; 
    }
}

File: 3248. Snake in Matrix.cpp
class Solution {
public:
    int finalPositionOfSnake(int n, vector<string>& commands) {
        int i = 0, j = 0;
        for(int k = 0; k < commands.size(); k++){
            if(commands[k] == "DOWN"){
                j++;
            }
            else if(commands[k] == "UP"){
                j--;
            }
            else if(commands[k] == "LEFT"){
                i--;
            }
            else{
                i++;
            }
        }
        int ans = n*j + i;
        return ans;
    }
};